lib/health.js

```diff
@@ -65,36 +70,103 @@ class HealthCheck {
     };
   }
 
-  addCheck (componentName, metricName, checkerPromise) {
+  /**
+   * Add a new health checking routine - for a metric, on a component
+   *
+   * @param {string} componentName - name of the component whose health is being
+   * reported
+   * @param {string} metricName - name of the metric on a component the value of
+   * which is being reported
+   * @param {AsyncFunction or Promise} checkExecutor - an async function to call
+   * for getting values or a Promise returning values.
+   * @param {int} minCacheMs - min cache duration in milliseconds.
+   */
+  addCheck (componentName, metricName, checkExecutor, minCacheMs=0) {
```

Comment:
If there is any possibility that this will continue to grow options, this pattern should change now.

`addCheck (componentName, metricName, checkExecutor, { minCacheMs: 0 }={}) { `

or leaving addCheck as the simple version with only three arguments, and adding a new 

`addCheckFromOpts ({componentName, metricName, checkExecutor, minCacheMs: 0}={}) { `

Using `addCheck` for the name of the new one is impossible while keeping the args nice for IDEs and keeping compatibility with current master branch, sadly.

Reply:
Great point and yes - very likely we may want other options in the future, so turning the fourth parameter into opts: 7d24e7e